apartmenttherapy/oxygen-lib-ui
# Henry/css placeholder

#### What does this PR do & why?
This turned into a much more lib-ui heavy approach than I originally anticipated. I was going to add placeholder styling into each variant's css file, but this was certainly faster for me. 

Replaces the placeholder image in each teaser with a component that renders and empty div and a `<style>` block that uses the teaser's picture's dimensions to style the block into the correct size and shape. 

#### How can this be manually tested?
I recommend grabbing oxygen https://github.com/apartmenttherapy/oxygen/pull/1423 if you actually want to see the placeholder, otherwise it's transparent.

Take a look at any collection page and you should see that each teaser has a correctly-sized area blocked off for the image before the image loads. It will help to throttle the connection, or just scroll down the page really fast. 

If you change your browser size the placeholders should resize appropriately.

#### Any questions or concerns about this change?
So soooooometimes, more than one teaser placeholder gets the same instance `_id` counter somehow. I think it's related to bundle loading and when the placeholder component is being used. I believe that a more random/unique id for each instance could fix this.

This adds a ton of `<style>` blocks to the markup. I hope the trade-off with not needing the image request will make up for this. Each SpreadBundle variant does have its own config file for images (e.g. `variantSoundbar.js`), so there's probably a way to set styles for the placeholders at the bundle level dynamically. If we don't get some speed wins with _this_ change, maybe that could be the next step.

#### GIF
![parking-spot-saver-360](https://user-images.githubusercontent.com/5115651/57955039-b2c8bf80-78c2-11e9-9ae3-facd8fbafded.gif)


#### Have you remembered to...?
